src/HasEnums.php

```diff
@@ -89,9 +182,39 @@ protected function getEnumClass(string $key): string
      */
     protected function asEnum(string $class, $value): Enumerable
     {
+        if ($value instanceof Enumerable) {
+            return $value;
+        }
+
         return forward_static_call(
-            $class.'::make',
+            $class . '::make',
             $value
         );
     }
+
+    /**
+     * @param Builder $builder
+     * @param string $method
+     * @param string $key
+     * @param int|string|Enumerable|int[]|string[]|Enumerable[] $enumerables
+     */
+    protected function buildEnumScope(
```

Comment:
In https://github.com/Astrotomic/laravel-translatable we have a similar approach to reduce the code duplicates and unify the logic.